sunkit_image/granule.py

```diff
@@ -0,0 +1,331 @@
+
+    segmented_image = segmented_image
+    segmented_image_fixed = np.copy(segmented_image.astype(float))
+    labeled_seg = skimage.measure.label(segmented_image + 1, connectivity=2)
+    values = np.unique(labeled_seg)
+    fac_count = 0
+    for value in values:
+        mask = np.zeros_like(segmented_image)
+        mask[labeled_seg == value] = 1
+        # check that is a 1 (white) region
+        if np.sum(np.multiply(mask, segmented_image)) > 0:
+            region_size = len(segmented_image_fixed[mask == 1])
+            tot_flux = np.sum(data[mask == 1])
+            # check that region is small
+            if region_size < fac_pix_limit:
+                # check that avg flux very high
+                if tot_flux / region_size > fac_brightness_limit:
+                    segmented_image_fixed[mask == 1] = 1.5
+                    fac_count += 1
+
+    gran_count = len(values) - 1 - fac_count  # subtract 1 for IG region
+
+    return segmented_image_fixed, fac_count, gran_count
+
+
+def kmeans_segment(data, llambda_axis=-1):
+    """
+    kmeans clustering: uses a kmeans algorithm to cluster data, in order to
+    independently verify the skimage clustering method (e.g using
+    cross_correlation() below).
+
+    Parameters
+    ----------
+    data : `numpy array`
+        Data to be clustered.
+    llambda_axis : `int`
+        Index for wavelength, -1 if scalar array.
+
+    Returns
+    -------
+    labels : `numpy.ndarray`
+        An array of labels, with 0 = granules, 2 = intergranules,
+        1 = in-between.
+    """
+
+    if llambda_axis not in [-1, 2]:
+        raise Exception(
+            "Wrong data shape. \
+        (either scalar or (x, y, llambda) )"
+        )
+    n_clusters = 3
+    n_init = 20
+    x_size = np.shape(data)[0]
+    y_size = np.shape(data)[1]
+    if llambda_axis == -1:
+        data_flat = np.reshape(data, (x_size * y_size, 1))
+        labels_flat = KMeans(n_clusters, n_init).fit(data_flat).labels_
+        labels = np.reshape(labels_flat, (x_size, y_size))
+    else:
+        llambda_size = np.shape(data)[llambda_axis]
+        data = np.reshape(data, (x_size * y_size, llambda_size))
+        labels = np.reshape(KMeans(n_clusters, n_init).fit(data), (x_size, y_size))
+
+    # make intergranules 0, granules 1:
+    group0_mean = np.mean(data[labels == 0])
+    group1_mean = np.mean(data[labels == 1])
+    group2_mean = np.mean(data[labels == 2])
+
+    # intergranules
+    min_index = np.argmin([group0_mean, group1_mean, group2_mean])
+    segmented_map = np.ones(labels.shape)
+
+    segmented_map[[labels[:, :] == min_index][0]] -= 1
+
+    return segmented_map
+
+
+def cross_correlation(segment1, segment2):
+    """
+    Return -1 and logging.info a message if the agreement between two arrays is
+    low, 0 otherwise. Designed to be used with segment and segment_kmeans
+    function.

```

Comment:
```suggestion
	Works out the cross correlation of two segmented arrays.
	
	-1 if the agreement between two arrays is low, 0 otherwise.
	
	Designed to be used with `segment` and `segment_kmeans` function.
```